src/Components/CreateImageWizardV2/CreateImageWizard.tsx

```diff
@@ -147,6 +150,29 @@ const CreateImageWizard = () => {
     useState<GCPAccountTypes>('googleAccount');
   const [gcpAccountEmail, setGcpAccountEmail] = useState('');
   const [gcpDomain, setGcpDomain] = useState('');
+  // Azure
+  const [azureManual, setAzureManual] = useState(false);
+  const [azureSource, setAzureSource] = useState<[number, string]>([0, '']);
+  const {
+    tenantId,
+    subscriptionId,
+    resourceGroups,
+    isError: azureIDError,
+  } = useGetAccountData(azureSource[0], 'azure');
```

Comment:
I think that this is the wrong place to call RTKQ hooks. `useGetAccountData` is not from RTKQ directly of course, but it wraps an RTKQ hook so the same argument applies. RTKQ hooks should be called as close as possible to where their return values are needed.

RTKQ hooks cache their values after fetching data, so they can be called wherever their values are needed. That means we don't need to pass down props like `isError` which is hard to keep track of... instead, we can just pass down the query parameters (so azureSource[0] and 'azure' in this case) and call the hook exactly where it is needed. This minimizes the number of props we need to keep track.

Second, calling hooks in the highest level component is potentially inefficient. It means that we front-load all of the data fetching. We may end up not needed to call some RTKQ hooks. Calling them where their data is needed ensures they only fetch data when necessary.

We do of course want the wizard to be fast so we do sometimes want to fetch the data ahead of time... but in that case, we should use prefetching and avoid passing down props:
https://redux-toolkit.js.org/rtk-query/usage/prefetching

Reply:
Yeah I forgot about this caching. Thanks for bringing it up!